lapack/tpls/KokkosLapack_gesv_tpl_spec_decl.hpp

```diff
@@ -390,6 +390,151 @@ KOKKOSLAPACK_CGESV_MAGMA(Kokkos::LayoutLeft, Kokkos::CudaSpace, false)
+        reinterpret_cast<cuDoubleComplex*>(Workspace.data()), IPIV.data(),
+        info.data()));
+
+    KOKKOS_CUSOLVER_SAFE_CALL_IMPL(cusolverDnZgetrs(
+        s.handle, CUBLAS_OP_N, m, nrhs,
+        reinterpret_cast<cuDoubleComplex*>(A.data()), lda, IPIV.data(),
+        reinterpret_cast<cuDoubleComplex*>(B.data()), ldb, info.data()));
+  }
+  KOKKOS_CUSOLVER_SAFE_CALL_IMPL(cusolverDnSetStream(s.handle, NULL));
+}
+
+#define KOKKOSLAPACK_GESV_CUSOLVER(SCALAR, LAYOUT, MEM_SPACE, ETI_SPEC_AVAIL)  \
+  template <>                                                                  \
+  struct GESV<                                                                 \
+      Kokkos::Cuda,                                                            \
+      Kokkos::View<SCALAR**, LAYOUT, Kokkos::Device<Kokkos::Cuda, MEM_SPACE>,  \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged>>,                   \
+      Kokkos::View<SCALAR**, LAYOUT, Kokkos::Device<Kokkos::Cuda, MEM_SPACE>,  \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged>>,                   \
+      Kokkos::View<int*, LAYOUT, Kokkos::Device<Kokkos::Cuda, MEM_SPACE>,      \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged>>,                   \
+      true, ETI_SPEC_AVAIL> {                                                  \
+    using AViewType = Kokkos::View<SCALAR**, LAYOUT,                           \
+                                   Kokkos::Device<Kokkos::Cuda, MEM_SPACE>,    \
+                                   Kokkos::MemoryTraits<Kokkos::Unmanaged>>;   \
+    using BViewType = Kokkos::View<SCALAR**, LAYOUT,                           \
+                                   Kokkos::Device<Kokkos::Cuda, MEM_SPACE>,    \
+                                   Kokkos::MemoryTraits<Kokkos::Unmanaged>>;   \
+    using PViewType =                                                          \
+        Kokkos::View<int*, LAYOUT, Kokkos::Device<Kokkos::Cuda, MEM_SPACE>,    \
+                     Kokkos::MemoryTraits<Kokkos::Unmanaged>>;                 \
+                                                                               \
+    static void gesv(const Kokkos::Cuda& space, const AViewType& A,            \
+                     const BViewType& B, const PViewType& IPIV) {              \
+      Kokkos::Profiling::pushRegion("KokkosLapack::gesv[TPL_CUSOLVER," #SCALAR \
+                                    "]");                                      \
+      gesv_print_specialization<AViewType, BViewType, PViewType>();            \
+                                                                               \
+      cusolverGesvWrapper(space, IPIV, A, B);                                  \
+      Kokkos::Profiling::popRegion();                                          \
+    }                                                                          \
+  };
+
+KOKKOSLAPACK_GESV_CUSOLVER(float, Kokkos::LayoutLeft, Kokkos::CudaSpace, true)
```

Comment:
Should the TPL support CudaUVMSpace as well?

Reply:
This is a question I am asking myself as well, right now I ETI for the following devices:

1. <Cuda, CudaSpace>
2. <HIP, HIPSpace>
3. <SYCL, SYCLDeviceUSMSpace>

If we want to ETI for the shared host/device memory space we need to double the ETI basically